scifight/utils.py

```diff
@@ -0,0 +1,34 @@
+
+def shorten_text(text, maxchars, ellipsis='...'):
+    """ Returns first few words from the `text`, but no more than `maxchars`, and appends
+        an `ellipsis`. This function works only on plain text (no HTML!), removes any leading
+        and trailing spaces, and squashes any number of consequent whitespaces (including
+        newlines) into a single space.
+
+        :param text: Input string, possibly quite long.
+        :param maxchars: Maximum length or result string.
+        :param ellipsis: Trailing sign to indicate implied continuation.
+        :return: Shortened substring with squashed whitespaces.
+    """
+    text = text.strip().replace('\n', ' ')
+    new_text = ''
+    have_space = False
+    for char in text:
+        if char == ' ':
+            if not have_space:
+                new_text += char
+                have_space = True
+        else:
+            new_text += char
+            have_space = False
+        if len(new_text) > maxchars:
+            break
+    text = new_text[:]
```

Comment:
Why make a copy? You `new_text` is a local variable and isn't accessed after this point anymore.